### PR TITLE
Fixes runtime with sound emotes

### PR DESCRIPTION
**Pull request by Adrer**
## What Does This PR Do
Fixes a runtime where emotes with the get_sound datum overridden would runtime when mobs that can't even run the emote tried to run it (Eg. Cat calling the *bark emote)
Fixes: #18023

## Why It's Good For The Game
Runtime bad

## Testing
Be a gorilla
Attempt to use the *sneeze emote, no runtime occurs
Attempt to use the *ooga emote, ooga loudly with no runtimes
Return to monke

## Changelog
No player facing changes.

### CODE REVIEW

File: code/modules/mob/mob_emote.dm [hunk 1]
```diff
@@ -28,12 +28,16 @@
...
 	for(var/datum/emote/P in key_emotes)
-		if(!P.check_cooldown(src, intentional))
-			// if an emote's on cooldown, don't spam them with messages of not being able to use it
-			silenced = TRUE
+		// can this mob run the emote at all?
+		if(!P.can_run_emote(src, intentional))
 			continue
-		if(P.try_run_emote(src, param, type_override, intentional))
-			return TRUE
+		else
```
Comment (Henri215) [inline code/modules/mob/mob_emote.dm:34]: That ```else``` is necessary?

Comment (Adrer) [reply code/modules/mob/mob_emote.dm:34]: No, purely added for readability, can be removed if preferred.

File: code/modules/mob/mob_emote.dm [hunk 2]
```diff
@@ -28,6 +28,9 @@
 		return FALSE
 	var/silenced = FALSE
 	for(var/datum/emote/P in key_emotes)
+		// can this mob run the emote at all?
+		if(!P.can_run_emote(src, intentional))
+			continue
```
Comment (SteelSlayer) [inline code/modules/mob/mob_emote.dm:33]: This is already ran as a part of the `if(P.try_run_emote` line below, so I'd rather avoid calling that twice. I think its probably fine to remove the one inside `try_run_emote`, and keep this one. What do you think @lewcc?

Comment (lewcc) [reply code/modules/mob/mob_emote.dm:33]: I agree, looks like you can remove it from there.

**Review by lewcc — CHANGES_REQUESTED** [review]
Please address steel's change

**Comment by Adrer** [comment]
> Please address steel's change
Requested change has been adressed

**Review by SteelSlayer — APPROVED** [review]

**Pull request merged**